rome hackathon blog (#4357)
PostHog/posthog.com: contents/blog/rome-hackathon.md

---
date: 2022-10-05
title: "All the cool things we built at our Rome hackathon"
rootPage: /blog
sidebar: Blog
showTitle: true
hideAnchor: true
author: ["ian-vanagas"]

> featuredImage: ../images/blog/product-people.png

Reviewer: ```suggestion
featuredImage: ../images/blog/planning-a-company-offsite/planning-offsite.jpeg
```

featuredImageType: full

> categories: ["Inside PostHog"]

Reviewer: ```suggestion
categories: ["Inside PostHog", "Engineering"]
```

---

> As an all-remote team, we know how important getting together in person is (we’ve [written about this before](/blog/asynchronous-remote-companies)). Our [product analytics team](/handbook/people/team-structure/product-analytics) (plus Luke, Annika, Cory, and Lottie) recently got back from an offsite in Rome. On top of eating a lot of pizza and gelato, refocusing and aligning work, and 360 feedback sessions, they had a hackathon and built a bunch of new things. 

Reviewer 2: Is it worth reorganising the intro so we have everything not about the hackathon _and_ then the bits about the hackathon. The "one outcome of the offsite...." para feels out of order

Reviewer: I'd probably go a bit further and suggest removing the "outcomes" from the intro and creating a conclusion around that theme which goes into a little more depth of what the objective actually means, rather than just linking to the handbook entry.

Author: Got it, moved to the conclusion and rewrote it.

> As an all-remote team, we know how important getting together in person is (we’ve [written about this before](/blog/asynchronous-remote-companies)). Our [product analytics team](/handbook/people/team-structure/product-analytics) (plus Luke, Annika, Cory, and Lottie) recently got back from an offsite in Rome. On top of eating a lot of pizza and gelato, refocusing and aligning work, and 360 feedback sessions, they had a hackathon and built a bunch of new things. 

Reviewer: Can we add a photo / photos of the team at the hackathon into the intro. Will add to the behind the scenes feel. We can also link / embed Paul's Twitter thread to add more color.

> We are showcasing them here. Some are sneak previews, others are works-in-progress, and they’ll likely change. We wanted to give you an idea of what our team actually does at an offsite hackathon.

Reviewer: Would be good to add a way people can give feedback here

Author: What is the recommended way to do this?

Author: Added Slack link, anything else?

Reviewer: Added a bit about raising an issue in the repo

One outcome of the offsite was a new focus on [nailing data exploration](handbook/people/team-structure/product-analytics#objective-1-nail-data-exploration). Many of the projects built related to this objective (on top of being fun and sparking joy). 

Here’s what the team built at their Rome hackathon.

## Frontend injected apps

Have you ever wanted to add floating pineapples to every page on your site? Marius worked on a way to make this dream a reality. More specifically, he built the ability to inject code into your page through [posthog-js](https://posthog.com/docs/integrate/client/js).

![Pineapple mode](../images/blog/rome-hackathon/pineapple-mode.gif)

> To do so, PostHog app developers add a `web.ts` file with the code they wanted to inject through PostHog into their app. This enables app developers to add features like notification banners and custom forms (or pineapples). 

Reviewer: ```suggestion
To do so, PostHog app developers add a `web.ts` file with the code they want to inject through PostHog into their app. This enables them to add features like notification banners and custom forms (or pineapples). 
```

Read more details in the [PR here](https://github.com/PostHog/posthog/pull/12003).

## Feedback app

> As a more serious example of what the frontend injected apps look like Marius and Luke built a feedback app.

Reviewer: ```suggestion
As a more serious example of what the frontend injected apps look like, Marius and Luke built a feedback app.
```

---

> date: 2022-10-05

Reviewer: ```suggestion
date: 2022-10-06
```

> title: "All the cool things we built at our Rome hackathon"

Reviewer: ```suggestion
title: "Coming soon to PostHog... all the cool things we built at our Rome hackathon"
```

rootPage: /blog
sidebar: Blog
showTitle: true
hideAnchor: true
author: ["ian-vanagas"]
featuredImage: ../images/blog/planning-a-company-offsite/planning-offsite.jpeg
featuredImageType: full

> categories: ["Inside PostHog", "Engineering", "Product updates"]

Reviewer: ```suggestion
categories: ["Inside PostHog", "Startups"]
```

---

> As an all-remote team, we know how important getting together in person is (we’ve [written about this before](/blog/asynchronous-remote-companies)). Our [product analytics team](/handbook/people/team-structure/product-analytics) (plus Luke, Annika, Cory, and Lottie) recently got back from an offsite in Rome. The team ate a lot of pizza and gelato, refocused and aligned current work, ran 360 feedback sessions, and more (see our engineer Paul's [summary thread here](https://twitter.com/pauldambra/status/1576257069766037506)). On top of all of this, they had a hackathon and built a bunch of new things.

Reviewer: ```suggestion
As an all-remote team, we know how important getting together in person is (we’ve [written about this before](/blog/asynchronous-remote-companies)). Our [product analytics team](/handbook/people/team-structure/product-analytics) (plus Luke, Annika, Cory, and Lottie) recently got back from an offsite in Rome – see also [how we plan a killer offsite in 8 weeks](/blog/planning-a-company-offsite). The team ate a lot of pizza and gelato, refocused and aligned current work, ran 360 feedback sessions, and more (see our engineer Paul's [summary thread here](https://twitter.com/pauldambra/status/1576257069766037506)). On top of all of this, they had a hackathon and built a bunch of new things.
```

## Hedgehog mode

To increase the amount of fun in PostHog, Ben and Lottie worked on Hedgehog mode. Hedgehog mode spawns a hedgehog to hang out and walk around on your page. Internal testing shows it helps meet important team KPIs like increased joy, lower stress, and more hedgehog thoughts.

![Hedgehog mode](../images/blog/rome-hackathon/hedgehog-mode.gif)

Some day the hedgehog might help you identify popular areas and potential areas for creating actions, but for now they are just hanging out. See the [PR here](https://github.com/PostHog/posthog/pull/12059). 

## Takeaways

One outcome of the offsite was a new focus on [nailing data exploration](handbook/people/team-structure/product-analytics#objective-1-nail-data-exploration). The team felt they had good data visualization tools, but not enough for data exploration. For example, it is difficult to move quickly between different types of data (events, persons, groups), often requiring recreating the same filters for multiple queries.

Many of the projects related to this objective (on top of being fun and sparking joy).

![Rome offsite team](../images/blog/rome-hackathon/rome-hackathon.jpeg)

> We like to be transparent with what we build. The details are found in the linked PRs as well as more expansively on the teams page in [our handbook](/handbook/people/team-structure/team-structure). We do this because we value feedback a lot. For example, users have been asking for better data exploration tools which helped it become a priority. 

Reviewer: ```suggestion
We like to be transparent with what we build. The details are found in the linked PRs as well as more expansively on the teams page in [our handbook](/handbook/people/team-structure/team-structure). We do this because we value feedback a lot. For example, users have been asking for better data exploration tools, which helped it become a priority. 
```

> If you have more feedback about your use cases or what we're planning to build, let us know on [our community Slack](/slack).

Reviewer: ```suggestion
If you have more feedback about your use cases or what we're planning to build, let us know on [our community Slack](/slack) or [raise an issue](https://github.com/PostHog/posthog/issues) on our repo 
```

> If you have more feedback about your use cases or what we're planning to build, let us know on [our community Slack](/slack) or [raise an issue](https://github.com/PostHog/posthog/issues) on our repo 

Reviewer: ```suggestion
If you have more feedback about your use cases or what we're planning to build, let us know on [our community Slack](/slack) or [raise an issue](https://github.com/PostHog/posthog/issues) on our repo. 
```

---
date: 2022-10-06

> title: "Coming soon to PostHog... all the cool things we built at our Rome hackathon"

Reviewer: ```suggestion
title: "All the cool things we built at our Rome hackathon"
```

To do so, PostHog app developers add a `web.ts` file with the code they want to inject through PostHog into their app. This enables them to add features like notification banners and custom forms (or pineapples). 

Read more details in the [PR here](https://github.com/PostHog/posthog/pull/12003).

## Feedback app

As a more serious example of what the frontend injected apps look like, Marius and Luke built a feedback app.

The feedback app is injected into your site through PostHog. It can be modified quickly by changing the `web.ts` file, without having to ship changes to your site. On your site, it takes input from users and delivers responses to their own tab in PostHog.

![Feedback app](../images/blog/rome-hackathon/feedback-app.png)

You can find the first PR for the feedback app [here](https://github.com/PostHog/feedback-app/pull/1).

## FaceHog (HogBook? Hoggr?) notifications

A problem some PostHog users face is others changing their carefully crafted insights and feature flags. They can mess these up or completely ruin them, and the original creator wouldn’t have any idea. Worse still, users could delete them without notice.

> To limit this happening, Paul worked on “FaceHog,” a notification log of changes to insights, dashboards, feature flags, and experiments you have created.

Reviewer: ```suggestion
To limit this happening, Paul worked on “FaceHog”, a notification log of changes to insights, dashboards, feature flags, and experiments you have created.
```

> To limit this happening, Paul worked on “FaceHog,” a notification log of changes to insights, dashboards, feature flags, and experiments you have created.

Reviewer 3: Is the comma in `“FaceHog,”` expected?

Author: Yeah, I like to keep my punctuation within the quotes, but I realize it looks particularly awkward here.
